### ISSUE TITLE
Third-party dep updates

### ISSUE THREAD

**Issue by GaryElshaw** [Coeur/transmission#8]
Just keeping an up to date list.

1. libdeflate is now at 1.24 (4/25) https://github.com/ebiggers/libdeflate/releases/tag/v1.24 Transmission is on 1.2.0
2. fast_float is now at 8.0.2 (3/25) https://github.com/fastfloat/fast_float/releases/tag/v8.0.2 Transmission is on 6.1.3
3. wide-integer now has releases (2/25). Transmission last updated it, July 14, 2022. https://github.com/ckormanyos/wide-integer/releases/tag/v1.01
4. libpsl is now at 0.21.5 (1/25) but had subsequent pushes until May. Transmission last updated February, 2022. https://github.com/rockdaboot/libpsl/releases/tag/0.21.5
5. libnatpmp. Hasn't been updated in 8 years. Has no official releases, but had its last push in August 2024. https://github.com/miniupnp/libnatpmp
6. fmt is now at 11.2.0, May 2025 https://github.com/fmtlib/fmt/releases/tag/11.2.0
7. miniupnpc is now at 2.3.3, May 2025 https://github.com/miniupnp/miniupnp/releases/tag/miniupnpc_2_3_3 (*2.3.0+ are the only versions that will receive ongoing [security support](https://github.com/miniupnp/miniupnp/blob/master/SECURITY.md)
8. googletest (5/25) is now at 1.17.0. Importantly, "Future development will [require at least C++17](https://opensource.google/documentation/policies/cplusplus-support#c_language_standard)". https://github.com/google/googletest/releases/tag/v1.17.0
9. utfcpp latest version is 4.0.9 (Dec, 25). https://github.com/nemtrif/utfcpp/releases/tag/v4.0.9
10. rapidjson has had a regular stream of pushes (2/25) and was last updated in Transmission Oct, 23. https://github.com/Tencent/rapidjson/
11. rpavlik-cmake-modules most recent push was Feb, 25. Last updated in Transmission 2023 https://github.com/transmission/rpavlik-cmake-modules

**Comment by Coeur** [comment]
Charles is not participating anymore, and Mike is only doing one PR a day with his own priorities (that may catch things up, but that will take more than 1 month).
I'll not do new PRs until I see my Mac build PR merged.
Thanks nonetheless for pointing out at dependency updates.

**Comment by GaryElshaw** [comment]
I see him over on electron more often than not. Have they had any more discussion regarding tearfur having write access, it seems like an obvious solution. 

I saw your Mac build pr, and i get holding off.

**Comment by Coeur** [comment]
PRs done as of today list.

**Comment by GaryElshaw** [comment]
Yes, to everything you said. https://github.com/transmission/transmission/issues/7058#issue-2460688555 There was a user a year or so back who got angry, and got banned after accusing Charles of running his own fiefdom. I couldn't really see that at the time, but do now. Open source projects live and die because of the leadership personalities involved. Mike is an equal part of the problem. If it takes someone to declare a project being dead, and the reaction is to pay attention for a day, it's lost. 

Just my 2c. Unless they are willing to give tearfur write access, immediately, the project should be forked. Part of my thinking on this is one of my pain-points. Charles would constantly refactor everything, and bug fixes were always secondary. Nothing is more painful than making your contributors constantly rework a problem you have previously resolved. @Pentaphon

**Comment by Pentaphon** [comment]
I agree for the most part, @GaryElshaw but I would try to not upset the admins and I definitely would bring you back to the project if there was a way to get you and Charles to work more independently of one another. Giving Tearfur write access is all people really want at this point and if they don't budge on it then I am officially worried about the project. I still have hope that things can improve without anything drastic.

**Comment by GaryElshaw** [comment]
As i said above, 24 hours of merging, and stopping, is insufficient after weeks of inaction. Also, completely absent, the third maintainer of the project is silent. The primary maintainer has been silent for weeks, and as Coeur pointed out, regularly contributes to another [project](https://github.com/electron/electron/commits/main/). My personal belief? Charles lost interest in the project when tearfur offered fixes to his code.

Just by the by, in the three or so years i was involved with Transmission(?), i've never heard of [bcomnes](https://github.com/bcomnes). I don't remember this person filing an issue or creating a pr for the project. Adjudicating a project when you've never contributed to it should make your perspective irrelevant.

**Comment by Pentaphon** [comment]
>My personal belief? Charles lost interest in the project when tearfur offered fixes to his code. Regardless of my feelings, he hasn't engaged with anything in the project for several weeks. And there were the several weeks of absence before that until pushing 4.0.6. It's why i raised the question of leadership personalities. The impediment to Transmission is the maintainers.

@GaryElshaw I agree, This whole write access issue with Tearfur would mean nothing if one of the maintainers took an active leadership role and was actually around to address, at the very least, the top 50 out of the 100 PRs and then merge the most important ones but for some reason, yes, there is absolutely no presence in the majority of our PRs. They won't even pop their head in and say "I like/don't like this big change", which is bizarre to me. They don't use all the tools Github has to offer like topics, stickies, dependency automation, etc and worst of all they got rid of you, Gary at a time when Transmission needs contributors the most. I can't imagine doing that to a project. Even if I didn't get along with somebody 100%. I'd just say: "hey let's figure out how to get along for the sake of the project" and work at that to get over whatever bad blood we had before.

I really hope something changes soon.

**Comment by GaryElshaw** [comment]
@Coeur @tearfur Just fork it. I'm deadly serious, and would welcome it. If two out of three maintainers can't be bothered even engaging in the conversation, for weeks, they don't care. https://github.com/electron/electron/commits/main/

**Comment by Pentaphon** [comment]
> Just fork it. I'm deadly serious, and would welcome and support it.

This is pretty risky, though. Do we even have at least 2 people who can commit to maintaining a fork of an active project? Would we risk fracturing the community by creating such a fork? Would today's contributors prefer to contribute to a longtime project or a brand new fork? I'm all for forking a project that has been officially abandoned, but the 3 current maintainers have yet to jump ship completely.

I think Coeur's idea to split the project into libtransmission and Transmission so as to compartmentalize write access and follow the model of projects like libtorrent-rasterbar and rakshasa is a more realistic first step and we should push for that first.

**Comment by Coeur** [comment]
I do not have the time in my life to maintain a fork all by myself. It could be tried and with multiple people involved, it could be sustainable, but I prefer to believe **for now** in Pentaphon's wisdom (https://github.com/transmission/transmission/issues/7058#issuecomment-2283755857):
>I'm in favor of escalating efforts to get the Transmission admins to listen to reason and allow somebody to move the project along themselves when they are absent or uninterested.

**Comment by Pentaphon** [comment]
Thanks, @Coeur . I hope @GaryElshaw knows I mean no offense to him with my disagreement about forking the project right now, but I am cautious by nature and simply do not think forking can be considered until all other possible options have been exhausted. i think perhaps a good first step is to poll the users in discussions as to whether they want libtransmissjon split from the larger project.

**Comment by GaryElshaw** [comment]
> I do not have the time in my life to maintain a fork all by myself.

This was not my suggestion for you to do this singularly and alone, it's why i said i would support the fork in any way i can. I'm offering to help out with project communication (wholly absent), bug triage, as well as UX support. These are my professional skills.

To @Pentaphon's point, i'm not offended in the slightest. I always appreciate your thoughts, because you always have a future for the project at heart. I do too. But i don't think it's realistic in its current configuration anymore. 

There are three maintainers.

It took enormous pressure for Charles to get the latest 4.0 version of Transmission out the door after a significant absence. After that push, he has been wholly absent from reviewing or proposing pr's.

Mitch has occasionally weighed in on MacOS changes, but in my time in the project, it's been infrequent, and the MacOS client code has been entirely changed without his input.

Mike is focussed on the Qt and GTK clients and infrastructure. #6642 is an easy fix, from February. #4376 was me picking up a change, Mike, himself proposed, that was stuck in limbo, and has now been further stuck in limbo since December 2022.

Neither Charles or Mike have even engaged in the conversation regarding write access to the project for months, or the divisions @Coeur recently proposed. 

You have to remember that the maintainers are regularly asked for review, or to respond to reported issues - by name. They receive the same notification of their name on GitHub as i see and respond to yours.

I will bet significant money that if @tearfur was to join a fork of Transmission, Charles will become available.

**Comment by GaryElshaw** [comment]
Just fork it. Right now, is obvious after a weekend. 
![image](https://github.com/user-attachments/assets/07c5e41e-c54e-46de-8363-4c0b912a622a)

**Comment by Pentaphon** [comment]
@GaryElshaw the thing is that we can say "fork it" but there is no certainty that there's enough people who can work on a theoretical fork any faster than the current team can, plus we don't know if enough people out there will just trust a new fork for an active project. There's also the issue of whether trackers will whitelist a new fork as well. They need to make @Coeur a member and get 4.0.7 out the door ASAP. It is the end of Summer and people might not be focusing on work as much as usual to cut a release in such a short timespan though.

**Comment by GaryElshaw** [comment]
I'm going to delete my comments on this 'off topic' issue tomorrow. 

There were weeks of inactivity before @Coeur asked if the project is abandoned. In the three weeks since, *none* of the maintainers have even engaged with any question from the project community, whether it be rights, or what should be alarming, to have a client being actively blacklisted, and have a sense of exigency to fix it. 

It's a failure of leadership to have so much talent and ignore it.

**Comment by GaryElshaw** [comment]
Seriously? When the client is banned? FFS. 
![image](https://github.com/user-attachments/assets/ff0c40ba-7b4c-4e2e-956f-f562b4a87dfc)
![image](https://github.com/user-attachments/assets/db71116e-358c-4499-ba7d-aebcc7c6e06e)

**Comment by GaryElshaw** [comment]
> I do not have the time in my life to maintain a fork all by myself. It could be tried and with multiple people involved, it could be sustainable, but I prefer to believe **for now** in Pentaphon's wisdom ([transmission#7058 (comment)](https://github.com/transmission/transmission/issues/7058#issuecomment-2283755857)):
> 
> > I'm in favor of escalating efforts to get the Transmission admins to listen to reason and allow somebody to move the project along themselves when they are absent or uninterested.

I can't believe i didn't respond to this. If you do want to fork the project, i'm very happy to contribute my time.

**Comment by Coeur** [comment]
I moved to other things in my life.
ckerr may have an economic motivation to maintain Transmission (there is a donation link), but I don't have the motivation for a fork.

Obviously, in nearly 1 year waiting ([example post from April 2024](https://github.com/transmission/transmission/issues/6813#issue-2266871499)), the admins didn't address the concerns the community had.

**Comment by GaryElshaw** [comment]
I'm both sorry and happy you've moved on to other things. Even this latest binge of merges was obviously to only get a second beta released, and we're back in to a moribund state until the next beta, i'm guessing. 

What was clear in this last batch of updates is that libtransmission work is the focus, because none of the GUI clients got anything signficant, at all. Clearly, the Mac client is suffering, and none of the old mac contributors, with the exception of nevack, are around. Don't get me started that in the entire list, [#4178](https://github.com/transmission/transmission/pull/4178) is just about the only thing still not merged, and it was ready to go in November 2022. 

I don't believe the reason is financial, i think this is about power and control, because it's certainly not about the health of the project if you alienate not just your regular contributors, but give up your occasional contributors by putting their pr's into indefinite limbo. Let alone ignoring your entire community who only wanted the project to thrive, and the maintainers couldn't even acknowledge the community's concerns.

Wishing you all the best Coeur, it was *almost always* a pleasure working with you :-) If there is something you're working on and think i might be able to help, i'm around.

Edit: Yes, i'm still banned from the project, but found out i'm still able to edit previous issues of my own, meaning i could close this issue here, and [edit this](https://github.com/transmission/transmission/issues/2999).

**Comment by GaryElshaw** [comment]
> [@GaryElshaw](https://github.com/GaryElshaw) I agree, This whole write access issue with Tearfur would mean nothing if one of the maintainers took an active leadership role and was actually around to address, at the very least, the top 50 out of the 100 PRs and then merge the most important ones but for some reason, yes, there is absolutely no presence in the majority of our PRs. They won't even pop their head in and say "I like/don't like this big change", which is bizarre to me. They don't use all the tools Github has to offer like topics, stickies, dependency automation, etc and worst of all they got rid of you, Gary at a time when Transmission needs contributors the most. I can't imagine doing that to a project. Even if I didn't get along with somebody 100%. I'd just say: "hey let's figure out how to get along for the sake of the project" and work at that to get over whatever bad blood we had before.
> 
> I really hope something changes soon.

I was banned from the project because i PR'ed to be removed as a member. Charles perceived it as some sort of hissy-fit, from his perspective, supposedly me daring him to ban me because i wanted to resign from the project. It had nothing to do with resigning from the project. 

The reason i asked to be removed as a member is because it was meaningless to be a 'member' when i couldn't even tag issues that were being filed. I had zero ability to do anything for the project than any person stumbling into the project. Charles had previously said the inability to tag was *probably* because of the previous interim bans i had. I lost faith that any of this was true, instead it was a message to no longer believe Charles, and my contributions were either meaningless, expedient, or disposable.

I'm remarkably unapologetic for the things i said that had those interim bans, because it's important to have a robust discussion, and it's important to say aloud, something is wrong.

In the same way this pr is perpetually stuck https://github.com/transmission/transmission/pull/4178 @Coeur This is perpetually stuck. I literally just picked up the work that was [abandoned by one of the maintainers](https://github.com/transmission/transmission/pull/4376) who could have merged their own pr. I worked hard on making this a reality, creating individual icons that include the ability to build light and dark versions. I am still stunned that the stalled work of a maintainer, progressed, hasn't been championed. 

I check in every day with Transmission. And every day feels like a rebuke for me being stupid, that i still care. The reason i do is because of the enormous amount of time i have given this project, the people i respect and have learnt from and respect; but its leadership doesn't give a fuck about its contributors. @Pentaphon 

https://github.com/Coeur/swiftmission/issues/8#issuecomment-2716668913

**Comment by Pentaphon** [comment]
I must admit, Transmission is being held too close to the chest instead of making it more of a community project like qBittorrent seems to be doing these days with their PRs always in the low double digits despite being a bigger project with way more end users. I personally don't want qBittorrent to be the only prominent torrent client out there while Transmission fizzles out... 

I feel like Transmission should have been at 4.3.0 beta 2 by now but there seems to be year-long breaks in between spikes of activity. 

What will it take for Transmission to expand as a project? I have no idea, but it seems like something major has to occur with the current leadership because I don't see anybody being able to fork this project and I don't know how long mikedld and ckerr think they will be able to maintain control over this project...

**Comment by GaryElshaw** [comment]
You have a far more diplomatic perspectve than mine. The barrier to Transmission progressing is solely responsible with the decisions of its disengaged maintainers. The majority of the existing 100 pr's should have been merged into 4.0 or 4.1 long ago, and the longer they languish, the more likely they have conflicts, the author no longer cares, or worse, the contributor is forced to recognise their contribution was a waste of time when their sense of urgency to fix something meant nothing. 

Instead, we have the principal maintainer, who occasionally arrives, who doesn't review pr's, but instead arrives to create and merge their own pr's before others.

Significantly, no maintainer has done any work on a 5.0, a roadmap, a schedule, or asked for proposals.

The reason i've always had a focus on third-party dependencies, and why it's alarming to me, is because they are 'dependencies', and they're also relegated. Every single dependency is out of date.

I'm also unhappy with the direction Rukario is taking the web client, it all feels wrong, and undoing what i hoped to accomplish that took a year. A feeling for another day.

I just noticed this from a thread when i first started getting involved with the project. Something i wrote and Charles responding.

https://github.com/transmission/transmission/issues/2462#issuecomment-1021980658
